File: app/src/main/java/com/stevenfrew/beatprompter/event/LineEvent.java
package com.stevenfrew.beatprompter.event;

import com.stevenfrew.beatprompter.Line;

public class LineEvent extends BaseEvent
{
    public Line mLine;
    public long mDuration;

    public LineEvent(long eventTime,long duration)
    {
        super(eventTime);
        mPrevLineEvent=this;
        mDuration=duration;
    }

    protected void offset(long amount)
    {
        super.offset(amount);
        mLine.mYStartScrollTime+=amount;
        mLine.mYStopScrollTime+=amount;
    }


}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/song/event/LineEvent.kt
package com.stevenfrew.beatprompter.song.event

import com.stevenfrew.beatprompter.song.line.Line

/**
 * A LineEvent tells the event processor to advance the song to the next line.
 */
class LineEvent(
	eventTime: Long,
	val line: Line
) : BaseEvent(eventTime) {
	override fun offset(nanoseconds: Long): BaseEvent = LineEvent(eventTime + nanoseconds, line)
}